### PR TITLE
add svg traitlet, more examples

### PR DESCRIPTION
**Pull request by bollwyvl**
First off, 😍 the work.

This adds SVG output with `XMLSerializer`... seems to work ok! Not sure if this is what you were talking about in the google group thread.

I added [some](https://nbviewer.jupyter.org/github/bollwyvl/ipysankeywidget/blob/static-svg/examples/Exporting%20Images.ipynb) [examples](https://nbviewer.jupyter.org/github/bollwyvl/ipysankeywidget/blob/static-svg/examples/Linking%20and%20Layout.ipynb) (and some navigation between them).

### CODE REVIEW

**Comment by ricklupton** [comment]
Thanks! Looks great, though there are a couple of typos in saving SVG to a file -- could you fix them?
The extra examples are good.

I was hoping there might be a way that the widget itself could have an SVG representation, so that it would automatically work in nbviewer without needing to separately display `SVG(widget.svg)`, but it seems like this will be complicated since it would have to be updated after the widget instance has been displayed.

File: ipysankeywidget/sankey_widget.py [hunk 1]
```diff
@@ -65,3 +74,20 @@ def save_png(self, filename):
...
+    @observe("svg")
+    def _on_svg_data(self, change):
+        if change['type'] != 'change':
+            return
+        if self._auto_svg_filename:
+            self.save_svg(self._auto_png_filename)
+            self._auto_svg_filename = None
+
+    def save_svg(self, filename):
+        data = base64.decodebytes(bytes(self.png, 'ascii'))
```
Comment (ricklupton) [inline ipysankeywidget/sankey_widget.py]: `self.svg`

File: ipysankeywidget/sankey_widget.py [hunk 2]
```diff
@@ -65,3 +74,20 @@ def save_png(self, filename):
     def auto_save_png(self, filename):
         self._auto_png_filename = filename
         return self
+
+    @observe("svg")
+    def _on_svg_data(self, change):
+        if change['type'] != 'change':
+            return
+        if self._auto_svg_filename:
+            self.save_svg(self._auto_png_filename)
```
Comment (ricklupton) [inline ipysankeywidget/sankey_widget.py]: `self._auto_svg_filename`

File: ipysankeywidget/sankey_widget.py (hunk 1 above)
Comment (bollwyvl) [inline ipysankeywidget/sankey_widget.py]: yep, didn't need to do anything, really, to the data, so just passing it directly

File: ipysankeywidget/sankey_widget.py (hunk 2 above)
Comment (bollwyvl) [inline ipysankeywidget/sankey_widget.py]: good catch!

**Comment by bollwyvl** [comment]
> the widget itself could have an SVG representation

Ah, that's a longer con... I don't know what parts of the situation are missing: you might need to publish on `npm`, do some extra embedding steps, etc.

Any thoughts, @sylvaincorlay?

**Comment by ricklupton** [comment]
Thanks for the fixes.

**Comment by bollwyvl** [comment]
Thanks! Looking forward to seeing how it develops!

**Pull request merged**